.github/workflows/wheels.yml

```diff
@@ -0,0 +1,58 @@
+name: Build all
+
+on: [push, pull_request]
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: macos-10.15
+            name: mac
+            cibw:
+              build: "cp36* cp37* cp38*"
+          - os: macos-10.15
+            name: mac-arm
+            cibw:
+              arch: universal2
+              build: "cp39*"
+    env:
+      CIBW_BUILD: "${{ matrix.cibw.build || '*' }}"
+      CIBW_ARCHS_MACOS: "${{ matrix.cibw.arch || 'auto' }}"
+      CIBW_BEFORE_BUILD_MACOS: "brew install libomp"
+      CIBW_REPAIR_WHEEL_COMMAND_MACOS: ""
+      # due to package and module name conflict have to temporarily move it away to run tests
+      CIBW_BEFORE_TEST_MACOS: "mv {package}/qsimcirq /tmp"
+      CIBW_TEST_COMMAND_MACOS: "pytest {package}/qsimcirq_tests/qsimcirq_test.py && mv /tmp/qsimcirq {package}"
+    steps:
+      - uses: actions/checkout@v2
+
+      # Used to host cibuildwheel
+      - uses: actions/setup-python@v2
+
+      - name: Install cibuildwheel and twine
+        run: python -m pip install cibuildwheel==1.11.0
+
+      - name: Run C++ tests
+        run: bash build_tools/test_libs.sh
+
+      - name: Build wheels
+        run: python -m cibuildwheel --output-dir wheelhouse
+
+      - uses: actions/upload-artifact@v2
+        with:
+          path: ./wheelhouse/*.whl
+
+      - name: Publish package
```

Comment:
It seems like the preferred way of doing this is to have a separate workflow with the [`on: release` trigger](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#example-using-multiple-events-with-activity-types-or-configuration).

Reply:
I know it's possible to share data between jobs but how is it possible to persist built wheels between workflows?

Reply:
Instead of persisting the wheel between workflows, we can just rebuild the wheel in the release workflow. (I usually cut releases via the Github API, which IIUC separates the "release" and "push" events.)

Reply:
Yeah, that makes sense, I opened a new PR with the changes https://github.com/quantumlib/qsim/pull/379